### PR TITLE
genpolicy: Add support for seccompProfile field

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

- [x] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
- [x] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [x] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [x] genPolicy only: Ensured the tool still builds on Windows
- [x] genPolicy only: Updated sample YAMLs' policy annotations, if applicable

###### Summary 

Add support for seccompProfile k8 yaml field and disallow requests with non null seccomp fields

Helpful links:
- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#seccompprofile-v1-core

- https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-seccomp-profile-for-a-container

###### Test Methodology

Test run: 100% pass

### CODE REVIEW

**Comment by danmihai1** [comment]
Please move the rules.rego change into the first commit. That way we should be able to upstream the first commit, but not yet the second - because upstream is using different test files.

File: src/agent/samples/policy/yaml/configmap/pod-cm1.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ spec:
       image: "mcr.microsoft.com/aks/e2e/library-busybox:master.220314.1-linux-amd64"
       securityContext:
         privileged: true
+        seccompProfile:
```
Comment (danmihai1) [inline src/agent/samples/policy/yaml/configmap/pod-cm1.yaml]: Please remove this change from the first commit.

File: src/tools/genpolicy/src/pod.rs [hunk 2]
```diff
@@ -279,6 +279,18 @@ struct SecurityContext {
...
+    #[serde(skip_serializing_if = "Option::is_none")]
+    seccompProfile: Option<SeccompProfile>,
+}
+
+/// /// See Reference / Kubernetes API / Workload Resources / Pod.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+struct SeccompProfile {
+    #[serde(rename = "type")]
+    profile_type: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (danmihai1) [inline src/tools/genpolicy/src/pod.rs:292]: Add an empty line between fields.

File: src/tools/genpolicy/src/pod.rs [hunk 3]
```diff
@@ -279,6 +279,18 @@ struct SecurityContext {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     runAsUser: Option<i64>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    seccompProfile: Option<SeccompProfile>,
+}
+
+/// /// See Reference / Kubernetes API / Workload Resources / Pod.
```
Comment (danmihai1) [inline src/tools/genpolicy/src/pod.rs:287]: Malformed comment line.

**Review by danmihai1 — APPROVED** [review]

**Comment by sprt** [comment]
Can you resolve the conflicts?

**Pull request merged**